src/models/bracket_races.rs

```diff
@@ -259,20 +281,7 @@ impl BracketRace {
             }
         };
 
-        let outcome = match (p1, p2) {
-            (PlayerResult::Forfeit, PlayerResult::Forfeit) => Outcome::Tie,
-            (PlayerResult::Forfeit, _) => Outcome::P2Win,
-            (_, PlayerResult::Forfeit) => Outcome::P1Win,
-            (PlayerResult::Finish(p1t), PlayerResult::Finish(p2t)) => {
-                if p1t < p2t {
-                    Outcome::P1Win
-                } else if p1t > p2t {
-                    Outcome::P2Win
-                } else {
-                    Outcome::Tie
-                }
-            }
-        };
+        let outcome: Outcome = From::from((&p1, &p2));
```

Comment:
i tried this a couple ways - i sort of expected `From<(&T, &T)>` to imply `From<(T, T)>` but iirc i tried that and it didn't work. i think `From<&T>` implies `From<T>` but I guess it doesn't distribute to tuples.